# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/mzayan-bit/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane

I chose  Lane 2  Refresh / Content Opportunity Scoring because it focuses on helping content teams identify which pages should be reviewed first. Instead of manually inspecting thousands of pages, machine learning can rank pages that are most likely to benefit from a content refresh. This saves time, improves prioritization, and provides a data-driven way to allocate limited SEO resources.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

### Decision
Which content pages should be prioritized for review and refresh?

### Action
The SEO or content team reviews the highest-ranked pages and decides whether to update, expand, improve, or monitor them.

### Cost of a Wrong Recommendation
If the model recommends the wrong pages, the team may waste time improving pages that were already performing well while missing pages that genuinely needed attention. This reduces efficiency and may result in lost search traffic.

### Why Machine Learning?
Large websites can contain thousands of pages. Reviewing every page manually is expensive and time-consuming. Machine learning can rank pages according to their likelihood of needing attention, helping teams focus their efforts where they are most valuable.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Columns:", len(df.columns))

print("\nTrend Direction")
print(df["trend_direction"].value_counts())

print("\nAverage Impressions:", round(df["impressions_90d"].mean(), 2))
print("Average Sessions:", round(df["sessions_90d"].mean(), 2))

Total pages: 30000
Columns: 44

Trend Direction
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average Impressions: 5200.37
Average Sessions: 37.07


## Quick Look at the Data

The starter dataset contains **30,000 content pages** with **44 features** describing search visibility, engagement, and content characteristics.

A quick exploration shows that:

- The dataset contains **30,000 pages**.
- **16,262 pages** have a **downward trend**, suggesting many pages may require review.
- The average number of impressions over the last 90 days is **5,200.37**.
- The average number of sessions over the last 90 days is **37.07**.

These statistics suggest there is enough variation in page performance to build a ranking system that helps prioritize which pages deserve attention first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This project is intended as a decision-support tool rather than proof of causation. The analysis identifies pages that may deserve review based on observable search and engagement signals. It does not prove that refreshing a page will improve its search performance, nor does it predict Google's ranking algorithm. Final decisions should always include human review.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.